.github/workflows/build.yml

```diff
@@ -32,15 +36,18 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
+      - name: ensure zstd for cache  # this should be removed once the arm64 VM includes zstd
+        if: ${{ matrix.os == 'arm64-dirty' || matrix.os == 'arm64-secure' }}
+        run: |
+          sudo apt install -y zstd
+      - name: update linuxkit cache if available
+        uses: actions/cache@v3
+        with:
+          path: ~/.linuxkit/cache
+          key: linuxkit-${{ matrix.arch }}-${{ github.sha }}
```

Comment:
Small question here: in case of need to re-run the workflow (for example because of connectivity issues), will we hit the problems with the old cache? Probably it would be nice to add `github.run_number` here?

Reply:
I think that might create problems, rather than resolve them. I worked up the scenarios:

* fails during packages: no cache is created, so it will have a cache miss on next run and create it: ✅ 
* fails during eve, rerun all: packages will load the cache, which is good; eve rerun will use the same cache: ✅ 
* fails during eve, rerun just eve: will use the cached output: ✅ 

Either way, we have the desired behaviour. What happens if we add the run number?

* fails during packages: no cache is created, so it will have a cache miss on next run and create it: ✅ 
* fails during eve, rerun all: packages will have a cache miss (different run_number), so it will recreate all, a bit inefficient, but it works: inefficient ✅ 
* fails during eve, rerun eve: when packages ran, it had a different run number, so there is no cache, eve will fail ❌ 

Adding run number either makes it fail, or makes it a bit more inefficient. I cannot find a scenario where leaving it out fails. 🤷‍♂️